pruning/architecture/pruning_methods/iterators.py

```diff
@@ -0,0 +1,90 @@
+from typing import Generator
+
+from config.iterators import PruningIterator
+import numpy as np
+
+
+def iterative(start: float, end: float, step: float) -> Generator[float, None, None]:
+    """
+    Generates a sequence of values from start to end with a given step size.
+
+    Args:
+        start (float): The starting value of the sequence.
+        end (float): The ending value of the sequence.
+        step (float): The step size between consecutive values.
+
+    Yields:
+        float: The next value in the sequence.
+
+    """
+    num_steps = int((end - start) / step) - 1
+
+    if start != 0:
+        yield start
+
+    for _ in np.linspace(start, end, num_steps):
+        yield step
+
+
+def one_shot(start: float, end: float, step: float) -> Generator[float, None, None]:
```

Comment:
I think a verb is missing in the naming, maybe something like construct_logarithmic_iterator

Reply:
Changed to classes with better naming.